test/widgets/dialog_checks.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:checks/checks.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+/// In a widget test, check that showErrorDialog was called with the right text.
+///
+/// Checks for an error dialog matching an expected title
+/// and, optionally, matching an expected message. Fails if none is found.
+///
+/// On success, returns the widget's "OK" button.
+/// Dismiss the dialog by calling `tester.tap(find.byWidget(okButton))`.
+Widget checkErrorDialog(WidgetTester tester, {
+  required String expectedTitle,
+  String? expectedMessage,
+}) {
+  final alertDialogList = tester.widgetList<AlertDialog>(find.byWidgetPredicate((widget) {
```

Comment:
I think probably simpler than having a big predicate passed to `byWidgetPredicate` would be to first find the dialog widget, and then inspect its `title` and `content`.

After all, there should be only one AlertDialog at a time — if there's one that matches the test's expectations, but somehow also a second one, then we're happy for that to fail the test too.

That can look like this:
```dart
  final dialog = tester.widget<AlertDialog>(find.byType(AlertDialog));
  tester.widget(find.descendant(matchRoot: true,
    of: find.byWidget(dialog.title!), matching: find.text(expectedTitle)));
  if (expectedMessage != null) {
    tester.widget(find.descendant(matchRoot: true,
      of: find.byWidget(dialog.content!), matching: find.text(expectedMessage)));
  }
```

As a bonus, if the matching fails, the stack trace will now identify whether (a) there was no AlertDialog at all, (b) it lacked the expected title, or (c) it lacked the expected message.